### DISCUSSION TITLE
Can't download videos from soap2day

### DISCUSSION THREAD

**Discussion by podag-byte** [aclap-dev/video-downloadhelper#1246] [Help and Q&A]
### Your question

Hello! I am unable to download videos from soap2day, from any of their sources (mainly vidplay, viper, and upcloud). It will start downloading but will always fail.

Here's a link of the video i'm trying to download: https://soap2dayto.ac/the-real-housewives-of-melbourne/

Any input would be much appreciated! thanks in advance

### Addon & CoApp version and details

version: 9.1.0.44
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"e1ebf3187ea24c1f"}
platform: arm64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 182,
    "successfull_dl": 68,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.4169, Firefox 130.0 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20, although that is largely irrelevant here, as will soon become apparent.

I went to your Australian housewives page & was actually somewhat surprised I was not geoblocked.  Geoblocking seems to be a favorite thing for Oz web sites to do to me.  I tried to get your show with VDH at least a dozen times.  The download would start out going gangbusters, downloading at several million bytes per second.  But then it would slow to a crawl & eventually fail.  The most I ever got of this was something like 28%.  What I got was playable but it was only part of the show.

I have seen behavior like this on Pornhub.  A download starts.  It's running at a couple million bytes per second.  Then it just slows to a crawl.  I'm too impatient to just wait to see if it will pick back up.  I generally reload VDH, then reload the web page, then try the exact same download again.  After 2, 3, 4, sometimes more retries, the download starts & runs at a few million bytes per second steadily.  I've seen Pornhub run up to nearly 30 million bytes per second.  Eventually, the download completes successfully.  Always.  Pornhub does always eventually cooperate.

But your Oz site wasn't giving me such cooperation.  So I went to Plan B.  The Network Monitor gave me an HLS master manifest.  I put that through ffprobe, a step I am not bothering to post here because it's duplicated at the start of the log I am posting.

I monitored things on the Network tab of the Resource Monitor.  This download was very weird.  It started off great, then slowed down.  I saw it drop near 0 several times.  But it kept getting up off the mat & resuming.  It just took patience.  Lots of it.

[Autralian Housewives mp4 Log.txt](https://github.com/user-attachments/files/16971392/Autralian.Housewives.mp4.Log.txt)

You should inspect that log carefully.  It is full of errors.  Corrupt packets.  I'm amazed it actually completed successfully.  But that it did.  I imagine all the error recovery ffmpeg had to do caused the average download speed (reported near the end of the log) to be so slow.  The downloaded file played in VLC fine.  Well, almost fine.  I didn't actually sit & watch it.  I have to say this kind of show holds no interest for me.  I can't see what anybody would see in these things.  Whatever.  Taste is what it is.  I can't disparage anybody.  I prefer the opera I'm listening to at the moment.  Don Giovanni from 1990/4/14.  Levine conducting Furlanetto, Ramey, Vaness, Upshaw, other equally wonderful artists.  Far more enjoyable.  Check https://archives.metopera.org/MetOperaSearch/ if you want the full details.

In any case, VLC did play this housewives show.  But as I was skimming through it, it seemed to lock up every so often.  Looking at selected bits more carefully, there seemed to be many spots where the VLC time counter would continue but the video was frozen for several seconds.  I assume this is due to the large number of corrupt packets that ffmpeg dropped.  At the start of the download log, you can see where the ffprobe report says the duration is 00:55:24.84.  The ffprobe I did on the file after the download completed (at the end of the log) says the duration is 00:53:58.03.  That's almost a minute & a half short.  The full credits do appear to be included at the end of what got recorded, so it's not that the download stopped prematurely.  I put the discrepancy down to the large number of dropped packets.

It seems that whatever parameters @paulrouget Paul is using for ffmpeg inside VDH are not as tolerant as the ones I am using.  You can see what I'm using at the beginning of the log I posted above.  I'm not sure which parameters are being so tolerant, but it looks like that's the case.  It's not that VDH couldn't download this content after some tweaking.  I am using the same ffmpeg that VDH uses, so it's certainly technically possible.  But VDH is having trouble with this content.  I believe it's because of the great mass of corruption in the source material.  The source material is corrupted.  Even if VDH were made to work, the result would still have the glitches I observed.  The source material is damaged.  There's nothing any downloader could possibly do to repair that.  The web site ought to be using better tools to archive their programs.

I note that this content contains Timed_ID3 data.  This used to be a problem for VDH.  But ffmpeg doesn't know how to interpret such data, so it ignores it.  This is why VDH now also is not bothered by Timed_ID3 data.  VDH didn't used to use ffmpeg for downloading, but now it does so Timed_ID3 data should not cause any problems here.  I have no idea what Timed_ID3 data is.  Web searches are singularly uneducational on the subject.  But this junk data seems to be rather widespread around the web.  I'd like to see what it does, just once, out of curiosity.  I have never had a problem downloading content that had Timed_ID3 data, that is, once I was able to filter out the Timed_ID3 data.  So it looks to me like it's just useless bloat.  I will stick to that opinion until somebody can show me an example of Timed_ID3 data doing something.

Bottom line:  I think VDH ought to be able to download this.  The result will almost certainly be damaged, but VDH should not fail.  I got ffmpeg, the same ffmpeg VDH uses, to download this.  So VDH ought to be able to accomplish the same result.  Paul needs to look into this.

**Reply by podag-byte** [reply]
oh my gosh thank you so much for the detailed analysis and the time you spent on investigating the issue! it is truly and deeply appreciated. wow I never thought the file would be so corrupted that it'd be hard to download, and yes! sometimes the videos do download but more often than not the audio would be messed up especially toward the end of the video. apologies i don't really understand a lot of technical terms used but still thank you again for all your efforts put into helping a stranger online. 

and thank you for the opera suggestions! i haven't really listened to opera before but i will check it out. I'm not trying to start an argument I just want to point out what was pointed out to me that I never realized until it was relayed to me on why I l and many others love Housewives. Try to name one TV show on air currently that features and focuses on an ensemble of middle aged woman, it'd be quite hard. and i think the Housewives has become such a successful international franchise is because it fills that niche. Although it has often been (rightfully) reduced to trashy reality that fuels drama (which is very true), the genuine connections and hardships these women go through on successful franchises really has a special places in many people's hearts. but, thank you again for everything! now i know why the videos don't download properly now.
